File: Python/7py 2/median.py
"""
|--------------------------------------------------------------------------
| All Star Code Challenge #14 - Find the median
|--------------------------------------------------------------------------
|
"""

def median(array):
   arr = sorted(array)
   mid = int(len(array)/2)
   return (arr[mid-1] + arr[mid]) / 2 if len(array)%2==0 else arr[mid]


from statistics import median as median
from numpy import median


def median(array):
   array.sort()
   n = len(array)
   return (array[n // 2] + array[(n-1) // 2]) / 2


def median(array):
   ordered = sorted(array)
   n = len(array)
   i = n >> 1
   return (ordered[i] + ordered[n - i - 1]) / 2


print(median([1234,345,78]),345)
print(median([33,99,100,30,29,50]), 41.5)
print(median([3,50]), 26.5)

File: JavaScript/7js 3/median.js
/*
|--------------------------------------------------------------------------
| All Star Code Challenge #14 - Find the median
|--------------------------------------------------------------------------
|
*/

function median(array) {
   const arr = array.sort((a, b) => a - b);
   const len = array.length;
   if (len % 2 == 0) {
      return (arr[len / 2] + arr[len / 2 - 1]) / 2
   } else {
      return arr[Math.floor(len / 2)]
   }
}


function median(A) {
   return A.sort((a, b) => a - b).length % 2 == 1 ? A[~~(A.length / 2)] : (A[~~(A.length / 2)] + A[~~(A.length / 2) - 1]) / 2
}


function median(array) {
   const arr = array.sort((a, b) => a - b)
   return arr.length % 2 == 1 ? arr[(arr.length - 1) / 2] : (arr[arr.length / 2] + arr[arr.length / 2 - 1]) / 2;
}


function median(arr) {
   let set = arr.sort((a, b) => a > b)
   let mid = Math.floor(set.length / 2)
   return (set.length % 2) === 0 ? (set[mid - 1] + set[mid]) / 2 : set[mid]
}


console.log(median([3, 2, 1]), 2);
console.log(median([33, 99, 100, 30, 29, 50]), 41.5);